compiler_test.go

```diff
@@ -1419,6 +1448,128 @@ func TestCompiler_ReplaceBuiltinModule(t *testing.T) {
 	checkValues(sharedValues2, "sharedValues2", 20)
 }
 
+func TestCompiler_ConcurrentParallelExecution(t *testing.T) {
+	sharedValues := map[string]int{}
+
+	createBuiltin := func(vals map[string]int, tengoMapValue int64) map[string]tengo.Object {
+		return map[string]tengo.Object{
+			"set": &tengo.UserFunction{
+				Value: func(args ...tengo.Object) (tengo.Object, error) {
+					n, _ := tengo.ToString(args[0])
+					i, _ := tengo.ToInt64(args[1])
+					vals[n] = int(i)
+					return nil, nil
+				},
+			},
+			"get": &tengo.UserFunction{
+				Value: func(args ...tengo.Object) (tengo.Object, error) {
+					return &tengo.Map{Value: map[string]tengo.Object{
+						"Value": &tengo.Int{Value: tengoMapValue},
+					}}, nil
+				},
+			},
+		}
+	}
+
+	srcM1 := `
+	m := import("builtin")
+	export { set: m.set }
+	`
+
+	srcM2 := `
+	m := import("m1")
+	export { set: m.set }
+	`
+
+	srcWithConstructor := `
+    m := import("builtin")
+
+	globalModuleVariable := m.get().Value + 20 // should update after replace builtin module
```

Comment:
I can't see you modify this variable in a module's code. It looks like you init it once and then work with it only in RO mode.

Also, I don't see any closures that export a hashmap bound to a variable declared inside closure.